src/cassandra.go

```diff
@@ -104,14 +104,26 @@ func main() {
 
 // runMetricCollection will perform the metrics collection.
 func runMetricCollection(i *integration.Integration, jmxClient *gojmx.Client) error {
+	definitions := NewDefinitions()
+
+	// Extra configuration will be enabled just when "metrics" mode is
+	// explicitly set to avoid colliding with inventory CONFIG_PATH.
```

Comment:
what if they enable both Metrics and inventory at the same time? Are the two args collying? 

What happens without passing neither "metrics=true" or "inventory=true"?

```
	// configPathEnv is the environment variable used by the infrastructure-agent to pass extra configuration
	// to integrations.
configPathEnv = "CONFIG_PATH"
```

and

```
	ConfigPath             string `default:"/etc/elasticsearch/elasticsearch.yml" help:"Path to the ElasticSearch configuration .yml file."`

```

Reply:
if they pass explicitly metrics=true / inventory=true that means there are 2 instances => no conflict
if they don't pass any, means all data will be collected => metrics + events + inventory => conflict => we don't load extra_config

This is an unfortunate collision between variables, but those features are implemented since long, would be hard to change them now without affecting customers

Reply:
Can we pass the configuration in other ways? I find it a bit obscure that without metrics or inventory flag we do not load extra_config.

Could we point to a file somewhere else like in https://github.com/newrelic/nri-vsphere/blob/483371b57ab62e77ec68ae347cfe8f0254c73baf/internal/config/config.go#L39?

It is a bit unfortunate not to have the config as:
```
  config:
    exclude:
      metrics:
        - "*"
    include:
      metrics:
        - query.CASWriteRequestsPerSecond
        - query.CASReadRequestsPerSecond
```
but the config just not working in some cases can be obscure

Reply:
There are more examples where we point to other files: [nri-oracledb](https://github.com/newrelic/nri-oracledb/blob/master/src/oracledb.go#L36) and [nri-mssql](https://github.com/newrelic/nri-mssql/blob/master/src/args/argument_list.go#L27) points there to include additional configuration to execute custom queries.

Reply:
Would be pretty easy to add it to another file, but ... from the customer perspective it's weird to point in the configuration file to another configuration file. This is not exactly a collection file, but more like extra configuration.

For exactly those scenarios CONFIG_PATH was added in infrastructure_agent
Since this is a new beta feature I would say to add support for `OHI_CONFIG_PATH` to avoid this collision if you agree.

Reply:
You are right those are different use cases, pointing to another file was one possible way to avoid the collision but maybe changing the environment variable name is more suitable here.

Reply:
I've renamed the variable and will add an extra one in the agent

Reply:
what do you mean? Will the agent pass both `CONFIG_PATH ` and `OHI_CONFIG_PATH` ? or is it already configurable?

Reply:
Because, as far as I understand, if you set the environment variable `CONFIG_PATH` and the inventory enabled, the integration will start to check the wrong path for `cassandra.yaml`

Reply:
only if inventory is enabled and the config block is in the wrong instance section.

Reply:
I am not sure I am following. 😅  This should not be a legitimate config:
```
integrations:
- name: nri-cassandra
  config:
    exclude:
      metrics:
        - "*"
    include:
      metrics:
        - query.CASWriteRequestsPerSecond
        - query.CASReadRequestsPerSecond
  env:
    PORT: "7199"
    HOSTNAME: localhost
    USERNAME: testUser
    PASSWORD: testPassword
    REMOTE_MONITORING: "true"
    # CONFIG_PATH is commented out since it is the default by it is overwritten due to the presence of config.path
    # CONFIG_PATH: /etc/cassandra/cassandra.yaml
  interval: 30s
  inventory_source: config/cassandra
  labels:
    env: production
    role: cassandra
```
As far as I understand, according to the old behavior `CONFIG_PATH` default value is overwritten by the agent due to the presence of the config section.

- What would happen now? Is the agent's behavior going to change?

Reply:
Context added -> https://github.com/newrelic/infrastructure-agent/pull/1395/files